### ISSUE TITLE
Update request: gnome.pomodoro 0.24.1 → 0.25.1

### ISSUE THREAD

**Issue by Herschenglime** [NixOS/nixpkgs#314999]
- Package name: gnome-pomodoro
- Latest released version: 0.25.1

- Current version on the unstable channel: 0.24.1
- Current version on the stable/release channel: 0.24.1

- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@jtojnar - I saw that you merged the last update from a bot

The current version's main app still works but the extension component is broken on gnome 46.1.

I tried simply doing the version bump with the following overlay:

```nix
    # see "Overriding a package inside a scope" in overlay wiki page
    # gnome scope overrides:
    gnome = prev.gnome.overrideScope' (gfinal: gprev: {
      # update gnome pomodoro to work on 46
      pomodoro = let
      in gprev.pomodoro.overrideAttrs (oldAttrs: rec {
        pname = "gnome-pomodoro";
        version = "0.25.1";
        src = prev.fetchFromGitHub {
          owner = pname;
          repo = pname;
          rev = version;
          hash = "sha256-tvNZqOw0XQF+3tD5b44LimPMHkmbYTu5wTtQfLHCgzo=";
        };
      });
    });
```

But the build fails because its unable to patch meson-post-install.sh:
```
gnome-pomodoro> building '/nix/store/x0ppil5xd59w81yzw98j7z3skbrxz6w5-gnome-pomodoro-0.25.1.drv'
gnome-pomodoro> Running phase: unpackPhase
gnome-pomodoro> unpacking source archive /nix/store/zhf40gsili4mg3scfhkh6740y2n0ia8g-source
gnome-pomodoro> source root is source
gnome-pomodoro> Running phase: patchPhase
gnome-pomodoro> applying patch /nix/store/s716ayi7iprbi9f3jn8g24ybwq65y5mq-fix-schema-path.patch
gnome-pomodoro> patching file data/meson.build
gnome-pomodoro> Hunk #1 succeeded at 52 (offset 21 lines).
gnome-pomodoro> can't find file to patch at input line 18
gnome-pomodoro> Perhaps you used the wrong -p or --strip option?
gnome-pomodoro> The text leading up to this was:
gnome-pomodoro> --------------------------
gnome-pomodoro> |diff --git a/meson-post-install.sh b/meson-post-install.sh
gnome-pomodoro> |index bf4013a..c87fba4 100644
gnome-pomodoro> |--- a/meson-post-install.sh
gnome-pomodoro> |+++ b/meson-post-install.sh
gnome-pomodoro> --------------------------
gnome-pomodoro> File to patch: 
gnome-pomodoro> Skip this patch? [y] 
gnome-pomodoro> Skipping patch.
gnome-pomodoro> 1 out of 1 hunk ignored
gnome-pomodoro> patching file meson.build
gnome-pomodoro> Hunk #1 succeeded at 42 (offset 2 lines).
error: builder for '/nix/store/x0ppil5xd59w81yzw98j7z3skbrxz6w5-gnome-pomodoro-0.25.1.drv' failed with exit code 1;
```

I'm not sure if there's something wrong with my overlay or something that needs to be changed about the patch script. Would this happen if something changed upstream?

Thanks!

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Herschenglime** [comment]
It does indeed seem like the meson-post-install.sh script was removed in this commit: 
https://github.com/gnome-pomodoro/gnome-pomodoro/commit/ce4b085006dda342db9a563302f8ffdafb569f95

I'll try to modify the patch accordingly so it doesn't try to edit a non-existent file and make a PR if it works. If it doesn't though, I think I'd need some guidance as I'm not entirely sure of the reasons behind the changes in the patch.

**Comment by jtojnar** [comment]
On FHS dustros, package manager typically recompiles global
`/usr/share/glib-2.0/schemas/gschemas.compiled` file on installation. Since
we want Nix packages to work without installation, we compile the file for
each package during package build. But to avoid multiple packagesʼ schenas
colliding when added to `buildEnv` created symlink tree, we use different
directory per package (path contains derivation name). The purpose of the
patch is to facilitate this.

Normally, the move is handled automatically by `glib`ʼs setup hook in
`postInstall`, but since the project hardcodes the schema path and knows
nothing about our convention, we need to patch it (we do that in the
`meson.build` part of the patch).

Changing the install directory in `data/meson.build` is not strictly
necessary, as the aforementioned `glib` setup hook can handle that later.
In fact, it should be removed since the `meson.post_install()` function
that replaces the script does not support our convention either.

**Comment by ljuzig** [comment]
Any update on this? gnome.pomodoro is currently broken on the latest NixOS (due to GNOME 46).
Thank you.

**Comment by Herschenglime** [comment]
I haven't really had time to take a crack at it since the initial bug report since I started a new job and have been using my Gnome install very little. If you'd like to try your hand at fixing it and report back, please let me know! I have roughly zero packaging experience so it'll be a lot of trial end error if I were to attempt a fix.

**Comment by Herschenglime** [comment]
So I've made a few more attempts over the months trying to get it myself but just haven't been successful, likely due to my relative lack of Gnome and Nix development experience. In the time since I made this issue, the current version has bumped up to v0.26.0.

My current progress is as follows:
https://github.com/NixOS/nixpkgs/commit/27ae86adfd9ac1cfb07207cb27469d06429353dd#diff-0e4792ae56ee0b9bb6df3c99913c477173fec69accba91194d57e71e17bbbb1bL30

Currently, this commit still fails to build, as glib-compile-schemas attempts to open the glib-2.0/schemas folder, but it is not found. The full output log can be found here:

https://gist.github.com/Herschenglime/f23cee579d5c3fa6089975ecf1a65f2e

I also tried simply removing the patch to see if the nix-specific glib hooks would pick up the schemas and put them in the right place, but unfortunately the extension part cannot be enabled on a package built this way.

On v25.2, I had a different issue, where the glibPreFixupHook would attempt to mv the schemas folder to its nix location, but the operation would fail as there was already a subfolder with the same name that mv refused to overwrite. I tried temporarily disabling or editing the hook to see if that would help, but I didn't know enough nix to make it work.

Should I try disabling the updated meson.post_install() call to compile schemas and just patch in a manual schema compilation? Is there some tool in the nix glib library that I'm missing?

I'm at a loss - sorry to ping @jtojnar, but any advice would be greatly appreciated!
